### PR TITLE
Upgrade to python 3.11

### LINKED ISSUE #549: Unable to initialize SearchClient on Python 3.11
- Algolia Client Version: 2.6.2
- Language Version: 3.11

### Description

When trying to initialize the `SearchClient` on Python 3.11 (with `aiohttp` installed to enable async requests), the following exception occurs:

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.11/site-packages/algoliasearch/search_client.py", line 55, in create
    return SearchClient.create_with_config(config)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/algoliasearch/search_client.py", line 67, in create_with_config
    from algoliasearch.search_client_async import SearchClientAsync
  File "/usr/local/lib/python3.11/site-packages/algoliasearch/search_client_async.py", line 8, in <module>
    from algoliasearch.http.transporter_async import TransporterAsync
  File "/usr/local/lib/python3.11/site-packages/algoliasearch/http/transporter_async.py", line 6, in <module>
    from algoliasearch.http.requester_async import RequesterAsync
  File "/usr/local/lib/python3.11/site-packages/algoliasearch/http/requester_async.py", line 10, in <module>
    class RequesterAsync(Requester):
  File "/usr/local/lib/python3.11/site-packages/algoliasearch/http/requester_async.py", line 16, in RequesterAsync
    @asyncio.coroutine  # type: ignore
     ^^^^^^^^^^^^^^^^^
AttributeError: module 'asyncio' has no attribute 'coroutine'. Did you mean: 'coroutines'?
```

It seems like it's related to the deprecation of the `@asyncio.coroutine` decorator, as described here: https://docs.python.org/3.10/library/asyncio-task.html#asyncio.coroutine

### Steps To Reproduce

Install the python 3.11 interpreter, or use a python 3.11 docker image (I used this command to test the issue `$ docker run --rm -it python:3.11 bash`

Then install `algoliaclient` and `aiohttp`

```
$ pip install algoliasearch==2.6.2 aiohttp==3.8.3
```

And the run the following in a python shell:

```python
from algoliasearch.search_client import SearchClient

client = SearchClient.create('xxx', 'xxx')
```

### PR DESCRIPTION
**Pull request by salomvary**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | yes
| New feature?      | no    
| BC breaks?        | What's BC?
| Related Issue     | Fix #549  
| Need Doc update   | yes

## Describe your change

Switch to `async def` instead of the `@asyncio.coroutine` which was removed in Python 3.11.

## What problem is this fixing?

Make this library work in Python 3.11.

### CODE REVIEW

**Comment by salomvary** [comment]
I don't know what "Expected — Waiting for status to be reported" means for CircleCI but it would be great to run the test there. I only have a sync integration that I manually tested to be working, but there is a little more to testing this... :)

**Comment by shortcuts** [comment]
Hey, thanks for the PR! I've allowed CI to run on forks but it requires a commit to trigger it, feel free to push anything when you have time :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,16 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
...
 
+## [v3.0.0](https://github.com/algolia/algoliasearch-client-python/compare/v2.6.1...v3.0.0)
+
+### Fixed
+
+- Python 3.11 compatibility [#551](https://github.com/algolia/algoliasearch-client-python/pull/551)
+
+### Changed
+
+- Python 2.7 is no longer supported
```
Comment (shortcuts) [inline CHANGELOG.md:18]:
```suggestion
- Major version: Python 2.7 is no longer supported
```

File: algoliasearch/http/requester_async.py [hunk 2]
```diff
@@ -13,8 +13,8 @@ def __init__(self):
 
         self._session = None
 
-    @asyncio.coroutine  # type: ignore
-    def send(self, request):  # type: ignore
+    # type: ignore
```
Comment (shortcuts) [inline algoliasearch/http/requester_async.py:16]: are the leftover type ignore still necessary here?

**Review by shortcuts — COMMENTED** [review]
looks great, thanks!

**Pull request closed without merging**